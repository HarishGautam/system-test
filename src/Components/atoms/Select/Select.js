import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import '../../../../styles/components/src/atoms/_select.scss';

const propTypes = {
  id: PropTypes.string.isRequired,
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
  disabled: PropTypes.string || PropTypes.bool,
  options: PropTypes.any.isRequired,
  selectedOption: PropTypes.string,
  placeholder: PropTypes.node,
};

const Select = ({
  id,
  className,
  disabled,
  placeholder,
  options,
  selectedOption,
  name,
  ...others
}) => (
  <React.Fragment>
    <select
      id={id}
      name={name}
      className={classnames(className)}
      value={selectedOption}
      disabled={disabled}
      {...others}
    >
      {placeholder && <option value="">{placeholder}</option>}
      {options.map(opt => (
        <option key={opt.label} value={opt.value} data-attr={opt.dataAttribute}>
          {opt.label}
        </option>
      ))}
    </select>
  </React.Fragment>
);

Select.defaultProps = {
  disabled: false,
  selectedOption: '',
  placeholder: '',
  className: '',
  showArrow: false,
};

Select.propTypes = propTypes;

export default Select;
