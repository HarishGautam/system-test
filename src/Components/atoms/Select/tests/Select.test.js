import React from 'react';
import { shallow } from 'enzyme';
import Select from '../Select';

describe('Select Component', () => {
  let input = '';

  test('should render correctly', () => {
    input = shallow(<Select options={['Option 1', 'Option 2']} />);
    expect(input).toMatchSnapshot();
  });
  test('should render with placeholder', () => {
    input = shallow(
      <Select placeholder="test" options={['Option 1', 'Option 2']} />
    );
    expect(input).toMatchSnapshot();
  });
});
