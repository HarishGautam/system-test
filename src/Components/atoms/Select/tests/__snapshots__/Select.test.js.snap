// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component should render correctly 1`] = `
<Fragment>
  <select
    className=""
    disabled={false}
    value=""
  >
    <option />
    <option />
  </select>
</Fragment>
`;

exports[`Select Component should render with placeholder 1`] = `
<Fragment>
  <select
    className=""
    disabled={false}
    value=""
  >
    <option
      value=""
    >
      test
    </option>
    <option />
    <option />
  </select>
</Fragment>
`;
